### PR TITLE
feat: [CDS-82819]: Changes in App setup delegate task

### PR DESCRIPTION
**Pull request by namantalaycha10**
## Describe your changes
 Changes in App setup delegate task:
 To use Artifact from Artifact bundle for deployment type , Basic, Rolling, Blue-Green and Canary  Deployment

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55946)

### CODE REVIEW

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by namantalaycha10** [comment]
trigger ut8

**Comment by namantalaycha10** [comment]
trigger ut9

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasStepHelper.java [hunk 1]
```diff
@@ -1573,6 +1588,8 @@ public TaskChainResponse executeTasTask(Ambiance ambiance, StepBaseParameters st
     tasStepPassThroughData.getUnitProgresses().add(unitProgress.build());
     return tasStepExecutor.executeTasTask(tasManifestOutcome, ambiance, stepElementParameters,
         TasExecutionPassThroughData.builder()
+            .tasManifestOutcome(
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasStepHelper.java:1591]: is this required, or being used somewhere?

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRollingDeployStep.java [hunk 2]
```diff
@@ -245,6 +263,19 @@ public TaskChainResponse executeTasTask(ManifestOutcome tasManifestOutcome, Ambi
     List<String> routeMaps = tasStepHelper.getRouteMaps(executionPassThroughData.getTasManifestsPackage(),
         getParameterFieldValue(tasRollingDeployStepParameters.getAdditionalRoutes()));
     TasInfraConfig tasInfraConfig = cdStepHelper.getTasInfraConfig(infrastructureOutcome, ambiance);
+
+    ArtifactBundleDetails artifactBundleDetails = null;
+    if (tasManifestOutcome != null && tasManifestOutcome.getType().equals(TAS_MANIFEST)
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasRollingDeployStep.java:268]: does using `tasManifestOutcome` from here, handle the cases of overrides as well?

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasCanaryAppSetupStep.java [hunk 3]
```diff
@@ -222,6 +226,19 @@ public TaskChainResponse executeTasTask(ManifestOutcome tasManifestOutcome, Ambi
     Integer olderActiveVersionCountToKeep =
         new BigDecimal(getParameterFieldValue(tasCanaryAppSetupStepParameters.getExistingVersionToKeep()))
             .intValueExact();
+
+    ArtifactBundleDetails artifactBundleDetails = null;
+    if (tasManifestOutcome != null && tasManifestOutcome.getType().equals(TAS_MANIFEST)
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasCanaryAppSetupStep.java:231]: does using `tasManifestOutcome` from here, handle the cases of overrides as well?

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasBasicAppSetupStep.java [hunk 4]
```diff
@@ -221,6 +225,19 @@ public TaskChainResponse executeTasTask(ManifestOutcome tasManifestOutcome, Ambi
     Integer olderActiveVersionCountToKeep =
         new BigDecimal(getParameterFieldValue(tasBasicAppSetupStepParameters.getExistingVersionToKeep()))
             .intValueExact();
+
+    ArtifactBundleDetails artifactBundleDetails = null;
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasBasicAppSetupStep.java:229]: same Q as below

File: 930-delegate-tasks/src/main/java/io/harness/delegate/pcf/TasBlueGreenSetupTaskHandler.java [hunk 5]
```diff
@@ -173,6 +175,15 @@ protected CfCommandResponseNG executeTaskInternal(CfCommandRequestNG cfCommandRe
 
       artifactFile = downloadArtifactFile(blueGreenSetupRequestNG, workingDirectory, logCallback);
 
+      String artifactPath = artifactFile == null ? null : artifactFile.getAbsolutePath();
+      if (blueGreenSetupRequestNG.getArtifactBundleDetails() != null) {
```
[the remark this answers is not in the document]
Comment (namantalaycha10) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/pcf/TasBlueGreenSetupTaskHandler.java]: Done

**Comment by namantalaycha10** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasCanaryAppSetupStep.java (hunk 3 above)
Comment (namantalaycha10) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasCanaryAppSetupStep.java:231]: sure will take care in override PR

**Comment by namantalaycha10** [comment]
trigger ut8

**Comment by namantalaycha10** [comment]
trigger ut5

**Review by RishabhGupta34 — APPROVED** [review]

**Pull request merged**